lib/transactions/forms.py

```diff
@@ -51,7 +62,12 @@ def __init__(self, *args, **kw):
         super(UpdateForm, self).__init__(*args, **kw)
 
     def clean(self):
-        keys = set(self.data.keys()).difference(set(self.fields.keys()))
+        fields = self.fields.keys()
+        # If the provider is already set, don't allow it to be changed.
+        if self.old.get('provider'):
```

Comment:
Can this maybe raise a validation error? I don't like data that silently disappears. Or at least log a warning?